File: Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace angular_dotnet.Models
{
    [Table("vehicles")]
    public class Vehicle
    {

        public int id { get; set; }

        public int modelid { get; set; }

        public Model model { get; set; }

        public bool is_registered { get; set; }

        [Required]
        [StringLength(255)]
        public string contact_name { get; set; }
        [StringLength(255)]
        public string contact_email { get; set; }

        
        [Required]
        [StringLength(255)]
        public string contact_phone { get; set; }

        public DateTime last_update { get; set; }

        public ICollection<VehicleFeature> features{get; set;}
        public Vehicle()
        {
            features = new Collection<VehicleFeature>();
        }



    }
}

File: ClientApp/src/app/models/vehicle.ts
export interface KeyValuePair {
    id: number;
    name: string;
}

export interface Contact {
    name: string;
    phone: string;
    email: string;
}

export interface Vehicle {
    id: number;
    model: KeyValuePair;
    make: KeyValuePair;
    is_registered: boolean;
    features: KeyValuePair[];
    contact: Contact;
    last_update: string;
}
export interface SaveVehicle {
    id: number;
    modelid: number;
    makeid: number;
    is_registered: boolean;
    features: number[];
    contact: Contact;
}
